main.go

```diff
@@ -295,62 +300,6 @@ func maxElementsPerInterval(monitorInterval time.Duration, samplesPerSecond int,
 	return uint32(uint16(samplesPerSecond) * uint16(monitorInterval.Seconds()) * presentCPUCores)
 }
 
-func sanityCheck(args *arguments) exitCode {
-	if args.samplesPerSecond < 1 {
-		return parseError("Invalid sampling frequency: %d", args.samplesPerSecond)
-	}
-
-	if args.mapScaleFactor > 8 {
-		return parseError("eBPF map scaling factor %d exceeds limit (max: %d)",
-			args.mapScaleFactor, maxArgMapScaleFactor)
-	}
-
-	if args.bpfVerifierLogLevel > 2 {
-		return parseError("Invalid eBPF verifier log level: %d", args.bpfVerifierLogLevel)
-	}
-
-	if args.probabilisticInterval < 1*time.Minute || args.probabilisticInterval > 5*time.Minute {
-		return parseError("Invalid argument for probabilistic-interval: use " +
-			"a duration between 1 and 5 minutes")
-	}
-
-	if args.probabilisticThreshold < 1 ||
-		args.probabilisticThreshold > tracer.ProbabilisticThresholdMax {
-		return parseError("Invalid argument for probabilistic-threshold. Value "+
-			"should be between 1 and %d", tracer.ProbabilisticThresholdMax)
-	}
-
-	if !args.noKernelVersionCheck {
-		major, minor, patch, err := tracer.GetCurrentKernelVersion()
-		if err != nil {
-			return failure("Failed to get kernel version: %v", err)
-		}
-
-		var minMajor, minMinor uint32
-		switch runtime.GOARCH {
-		case "amd64":
-			if args.verboseMode {
-				minMajor, minMinor = 5, 2
-			} else {
-				minMajor, minMinor = 4, 19
-			}
-		case "arm64":
-			// Older ARM64 kernel versions have broken bpf_probe_read.
-			// https://github.com/torvalds/linux/commit/6ae08ae3dea2cfa03dd3665a3c8475c2d429ef47
-			minMajor, minMinor = 5, 5
-		default:
-			return failure("Unsupported architecture: %s", runtime.GOARCH)
-		}
-
-		if major < minMajor || (major == minMajor && minor < minMinor) {
-			return failure("Host Agent requires kernel version "+
-				"%d.%d or newer but got %d.%d.%d", minMajor, minMinor, major, minor, patch)
-		}
-	}
-
-	return exitSuccess
-}
-
 func parseError(msg string, args ...interface{}) exitCode {
```

Comment:
This help function was meant to reduce code duplication. Now, this function is only called once and thus have no purpose any more.